src/day09.rs

```diff
@@ -1,30 +1,364 @@
 // Potential improvements:
-//
+// Vec turns out to be borderline impossible
+// HashMap is much better but is stil quite slow
+// Doubly linked list with a cursor sounds best but rust only has these in nightly..
 
-pub fn day09(_input_lines: &[Vec<String>], _solution_ver: &str) -> (String, String) {
-    let answer1 = 0;
-    let answer2 = 0;
-    (format!("{}", answer1), format!("{}", answer2))
+// Rough runtimes:
+// Vec solves in something in the order of a few hours
+// HashMap solves in 2.5 to 3 seconds
+// LinkedList solves in 250 to 300 milliseconds
+use std::{collections::{HashMap, LinkedList, linked_list::CursorMut}, vec, fmt::Display};
+
+pub fn day09(input_lines: &[Vec<String>], solution_ver: &str) -> (String, String) {
+    let mut input = input_lines[0][0].split_ascii_whitespace();
+    let n_players = input.nth(0).unwrap().parse::<usize>().unwrap();
+    let final_marble = input.nth(5).unwrap().parse::<i32>().unwrap();
+    match solution_ver {
```

Comment:
The duplication and string matching here is a bit nasty. You could pass the `MarbleCollection` directly into this function making it generic, you may want to reuire that it implements default so that you can create as many as you want with `T::default()`.